player/lua/stats.lua

```diff
@@ -867,24 +867,90 @@ local function eval_ass_formatting()
     end
 end
 
+-- assumptions:
+--   s is composed of SGR escape sequences and/or printable UTF8 sequences
+--   printable codepoints occupy one terminal cell (we don't have wcwidth)
+--   tabstops are 8, 16, 24..., and the output starts at 0 or a tabstop
+-- note: if maxwidth <= 2 and s doesn't fit: the result is "..." (more than 2)
+function term_ellipsis(s, maxwidth)
+    local TAB, ESC, SGR_END = 9, 27, ("m"):byte()
+    local width, ellipsis = 0, "..."
+    local fit_len, in_sgr
+
+    for i = 1, #s do
+        local x = s:byte(i)
+
+        if in_sgr then
+            in_sgr = x ~= SGR_END
+        elseif x == ESC then
+            in_sgr = true
+            ellipsis = "\27[0m..."  -- ensure SGR reset
+        elseif x < 128 or x >= 192 then  -- non UTF8-continuation
+            -- tab adds till the next stop, else add 1
+            width = width + (x == TAB and 8 - width % 8 or 1)
+
+            if fit_len == nil and width > maxwidth - 3 then
+                fit_len = i - 1  -- adding "..." still fits maxwidth
+            end
+            if width > maxwidth then
+                return s:sub(1, fit_len) .. ellipsis
+            end
+        end
+    end
+
+    return s
+end
+
+local function term_ellipsis_array(arr, from, to, maxwidth)
+    for i = from, to do
+        arr[i] = term_ellipsis(arr[i], maxwidth)
+    end
+    return arr
+end
+
+-- split str into a table
+-- example: local t = split(s, "\n")
+-- init: where to start looking for the separator pattern (default 1)
```

Comment:
Even though the `init` comment is mine and it's not wrong, it's also misleading, because the behavior is as if `str` was `str:sub(init)` (I added this init comment quickly right before I posted this function. my bad).

Also, because `init` is never used by callers, and currently it's unlikely to be used, I suggest to remove the `init` argument and always make it `1` inside the function.

`plain` can stay because splitting on a pattern can be useful, even if currently unused.